app/models/bulkrax/csv_entry.rb

```diff
@@ -107,7 +107,9 @@ def build_export_metadata
     # Metadata required by Bulkrax for round-tripping
     def build_system_metadata
       self.parsed_metadata['id'] = hyrax_record.id
-      self.parsed_metadata[source_identifier] = hyrax_record.send(work_identifier)
+      source_id = hyrax_record.send(work_identifier)
+      source_id = source_id.to_a.first if source_id.is_a?(ActiveTriples::Relation)
+      self.parsed_metadata[source_identifier] = source_id
```

Comment:
in the event that `source_id` is not an  ActiveTriples::Relation, does that mean that sometimes `self.parsed_metadata[source_identifier]` will be a string and other times it will be another data type?

that feels weird to me.

Reply:
The value was being used as-is previously, so out of caution i took the approach of checking for Relation before transforming it just in case the previous behavior was applicable for some situation I wasn't aware of. I'm okay with just calling `source_id.to_a.first` if we are sure the value will always be a Relation (assuming there is not a misconfiguration and an unsuitable field is attempting to be used for the source_identifier, like an object field or something).

Reply:
my first thought was that this felt weird, but I've been looking at some code and things and I think this change might actually be ok.